### PR TITLE
[WIP]Configurable speed parameters

### PR DESCRIPTION
**Pull request by cua-cua**

### CODE REVIEW

File: src/speed.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+{
+	return turn_radius;
+}
+
+void set_turn_radius(float value)
+{
+	turn_radius = value;
+}
+
+float get_t0(void)
```
Comment (Peque) [inline src/speed.c]: I'd say `get_turn_t0`, for consistency.

Comment (Peque) [reply src/speed.c]: (same for `set_t0` and `*_t1` functions)

File: src/speed.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#ifndef __SPEED_H
+#define __SPEED_H
+
+#include "setup.h"
+
+/* Speed modes*/
```
Comment (Peque) [inline src/speed.h]: Missing space after `modes`.

File: src/speed.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#define SLOW_MODE 0
+#define NORMAL_MODE 1
+#define FAST_MODE 2
```
Comment (Peque) [inline src/speed.h]: I'd say define this as an `enum` or do not define those speed names at all.

File: src/target/move.c [hunk 4]
```diff
@@ -334,12 +322,16 @@ static void move_side(enum step_direction side)
...
-			0.02 - MOUSE_AXIS_SEPARATION / 2., 0.448);
+			CELL_DIMENSION / 2 - get_turn_radius() -
+			    MOUSE_AXIS_SEPARATION / 2.,
+			get_turn_linear_speed());
 	disable_walls_control();
 	turn_side(side);
 	enable_walls_control();
 	target_straight(get_encoder_average_micrometers(),
-			0.02 + MOUSE_AXIS_SEPARATION / 2., max_linear_speed);
+			-CELL_DIMENSION / 2 + get_turn_radius() +
```
Comment (Peque) [inline src/target/move.c]: Why changing the sign here? I think it should be `CELL_DIMENSION / 2 - get_turn_radius()`, just like in the `target_straight` call above.

File: src/speed.c [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+static volatile float t0;
+static volatile float t1;
+
+const float linear_acceleration_defaults[NUM_MODES] = {5., 0., 0.};
+const float linear_deceleration_defaults[NUM_MODES] = {5., 0., 0.};
+const float angular_acceleration_defaults[NUM_MODES] = {32. * PI, 0., 0.};
+const float max_linear_speed_defaults[NUM_MODES] = {.5, 0., 0.};
+const float max_angular_speed_defaults[NUM_MODES] = {3 * PI, 0., 0.};
+const float turn_linear_speed_defaults[NUM_MODES] = {0.448, 0., 0.};
+const float turn_radius_defaults[NUM_MODES] = {0.11, 0., 0.};
```
Comment (Peque) [inline src/speed.c]: Default should be `0.07.` I think, not `0.11`.

File: src/speed.c (hunk 1 above)
Comment (cua-cua) [reply src/speed.c]: ok

File: src/speed.c (hunk 5 above)
Comment (cua-cua) [reply src/speed.c]: :+1:

File: src/target/move.c (hunk 4 above)
Comment (cua-cua) [reply src/target/move.c]: ok

File: src/speed.h (hunk 3 above)
Comment (cua-cua) [reply src/speed.h]: I will not define them at the moment

**Pull request merged**